ArmRobot/Assets/Scripts/DomainRandomization.cs

```diff
@@ -0,0 +1,218 @@
+            listOfAlreadyMovedObjects.Add(gameobject); // we add the gameobject to the already moved objects 
+        }
+        
+        
+        // move robot 
+        MoveRobot(rotationAngle);
+        // then we change its color and the color of all its children
+        foreach (Transform child in robot.transform){
+            ColorRandomizer colorRandomizerChildRobot = child.gameObject.GetComponent<ColorRandomizer>();
+            colorRandomizerChildRobot.ChangeColor();
+        } 
+        
+        // move the camera
+        GameObject camera = GameObject.Find("VisionCamera");
+        RandomizerPositionCamera randomizedPositionCamera = camera.GetComponent<RandomizerPositionCamera>();
+        randomizedPositionCamera.Move();
+
+        // change the color of the table 
+        ColorRandomizer colorRandomizerTable = table.GetComponent<ColorRandomizer>();
+        colorRandomizerTable.ChangeColor();
+
+        // change the Directionalight
+        GameObject directionLight = GameObject.Find("DirectionalLight");
+        DirectionalLightRandomization directionalLightRandomizer = directionLight.GetComponent<DirectionalLightRandomization>();
+        directionalLightRandomizer.UpdateLight();
+    
+    }
+
+    public void MoveRobot(float rotationAngle)
+    {
+        // this function is designed to do the move operation for the robot
+        RobotController robotController = robot.GetComponent<RobotController>();
+        float rangeRotation = -rotationAngle + 2 * rotationAngle * Random.value;
+        float[] rotation = {rangeRotation, rangeRotation, rangeRotation, rangeRotation, rangeRotation, rangeRotation};
+        robotController.ForceJointsToRotations(rotation);
+    }
+
+    List<GameObject> CreateRandomListCylinders()
+    {
+        // this function is designed to create a list of a random number of cylinders and instantiate them 
+        float minimumDistanceBetweenObjects = scaleObject + minimumDistanceWithoutObjects;
+        int randomNumberCylinders = 1 + Random.Range(1, nbMaxCylinders); 
+        List<GameObject> listOfCylinders = new List<GameObject>();
+        for (var i = 0; i < randomNumberCylinders; i++) {
+            GameObject cylinder  = GameObject.CreatePrimitive(PrimitiveType.Cylinder);
+            
+            cylinder.transform.position = new Vector3(robotMaxReach - 2*minimumDistanceWithoutObjects , 
+                                                yAltitudeTable, robotMaxReach - 2*minimumDistanceWithoutObjects - minimumDistanceBetweenObjects*i);
+            
+            
+            cylinder.transform.localScale = new Vector3(scaleObject, scaleObject, scaleObject);
+            cylinder.AddComponent<ColorRandomizer>();
+            cylinder.AddComponent<RandomizerPositionObject>();
+            cylinder.tag = "Cylinder";
+            ColorRandomizer colorRandomizer = cylinder.GetComponent<ColorRandomizer>();
+            colorRandomizer.ChangeColor();
```

Comment:
It would be even more readable and modular to break out another method to create one cylinder, like `GameObject CreateCylinder(Vector3 position, Vector3 scale)` that we could call from this method.